### ISSUE TITLE
seccomp failure on Android trying to call openat2 from open_beneath

### ISSUE THREAD

**Issue by hrydgard** [bytecodealliance/cap-std#196]
I'm trying to run some code that uses cap-std on Android 10 (on a kernel that definitely does not support the syscall), and I hit the following crash in `openat2`, called by `open_beneath` from `open_impl`:

```
 signal 31 (SIGSYS), code 1 (SYS_SECCOMP), fault addr --------
 Cause: seccomp prevented call to disallowed arm64 system call 437

 backtrace:
       #00 pc 00000000000703d0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32) 
       #01 pc 0000000002e8e404  ourapp.so (rsix::imp::libc::syscalls::openat2::h04809da1ec63d9ef+44) 
       #02 pc 0000000002e8aa9c  ourapp.so (cap_primitives::rsix::linux::fs::open_impl::open_beneath::h3811b95d26505c84+312) 
       #03 pc 0000000002e8a800  ourapp.so (cap_primitives::rsix::linux::fs::open_impl::open_impl::h725b24260261d337+40)
       ...
```

Looking back up into cap-std, it seems that the detection method for presence of `openat2` is inadequate, because instead of simply returning ENOSYS as the linked code assumes, Android forcibly kills the process with a SIGSYS signal.

https://github.com/bytecodealliance/cap-std/blob/6f38d0a51a9dec8ed034fd6f57bc25e4bb2255f0/cap-primitives/src/rsix/linux/fs/open_impl.rs#L103

So there needs to be some sort of kernel version check instead I guess?

**Comment by sunfishcode** [comment]
Thanks for the report! I've now filed #197 which implements a kernel version check, which may fix the problem. I don't have an Android device to test it on though. Would you be able to test this fix?

**Comment by hrydgard** [comment]
Thank you! I will test it tomorrow.

**Comment by hrydgard** [comment]
I can confirm that your version check indeed solves the problem!

**Comment by sunfishcode** [comment]
Thanks! This is now released in 0.21.1.
